testing/extended/verify_env.py

```
@@ -0,0 +1,310 @@
+    ]
+    try:
+        # subprocess.run with a list avoids host-side shell interpretation.
+        # The remote_cmd argument is shlex-quoted above, so the SSH remote
+        # shell receives a safe command string. We accept env-derived inputs
+        # here only because (a) this is a developer-only test tool that
+        # reads ~/.claude/dev-secrets.env (operator-owned, not network-
+        # sourced), (b) every interpolated value is shlex-joined, and (c) a
+        # malicious env file is already game-over for the operator. The
+        # nosemgrep directive must be on the SAME line as the call — Semgrep
+        # OSS does NOT honor the comment-on-preceding-line form.
+        r = subprocess.run(cmd, capture_output=True, text=True, timeout=20)  # nosemgrep: python.lang.security.audit.dangerous-subprocess-use-tainted-env-args.dangerous-subprocess-use-tainted-env-args
```

Ruff (0.15.13)

[error] `subprocess` call: check for execution of untrusted input

(S603)